sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/models/LinkedEntity.java

```diff
@@ -3,50 +3,115 @@
 
 package com.azure.ai.textanalytics.models;
 
+import com.azure.core.annotation.Immutable;
 import com.azure.core.util.IterableStream;
 
 /**
  * The {@link LinkedEntity} model.
  */
-public interface LinkedEntity {
+@Immutable
+public final class LinkedEntity {
+    /*
+     * Entity Linking formal name.
+     */
+    private final String name;
+
+    /*
+     * List of instances this entity appears in the text.
+     */
+    private final IterableStream<LinkedEntityMatch> linkedEntityMatches;
+
+    /*
+     * Language used in the data source.
+     */
+    private final String language;
+
+    /*
+     * Unique identifier of the recognized entity from the data source.
+     */
+    private final String dataSourceEntityId;
+
+    /*
+     * URL for the entity's page from the data source.
+     */
+    private final String url;
+
+    /*
+     * Data source used to extract entity linking, such as Wiki/Bing etc.
+     */
+    private final String dataSource;
+
+    /**
+     * Creates a {@link LinkedEntity} model that describes linked entity.
+     *
+     * @param name The entity Linking formal name.
+     * @param linkedEntityMatches A list of instances this entity appears in the text.
+     * @param language The language used in the data source.
+     * @param dataSourceEntityId Unique identifier of the recognized entity from the data source.
+     * @param url URL for the entity's page from the data source.
+     * @param dataSource The data source used to extract entity linking, such as Wiki/Bing etc.
+     */
+    public LinkedEntity(String name, IterableStream<LinkedEntityMatch> linkedEntityMatches, String language,
```

Comment:
Should `linkedEntityMatches` also be changed to `matches` here (not sure if this is exposed to users, but would still be nice to keep it consistent)